apache/hive
# HIVE-27402: Refactor Kafka credential handling into kafka-handler module

### What changes were proposed in this pull request?
#### Production changes:
1. Add new internal interface DagCredentialSupplier in hive-exec module encapsulating and abstracting the token retrieval process.
2. Move Kafka token retrieval code from DagUtils (hive-exec) to KafkaDagCredentialSupplier (kafka-handler).
3. Remove now redundant dependency to kafka-client from hive-exec module.

#### Test changes:
4. Expose constructor in DagUtils to allow unit testing with custom credential suppliers.
5. Add hadoop-minikdc dependency in kafka-handler to allow Kerberized tests and centralize exclusions in the main pom.xml file.
6. Enhance KafkaBrokerResource to support SAML connections with Kerberos to enable testing the Kafka token retrieval logic.
7. Add tests for affected code to ensure it is functional (and will remain as such in the future).


### Why are the changes needed?
Currently the code for handling credentials and authentication for the Kafka storage handler is spread across hive-exec and kafka-handler modules. This leads to unexpected dependencies to kafka (e.g., kafka-client) in hive-exec module and limits sharing opportunities of credential utilities present in KafkaUtil class.

The goal of this refactoring is threefold:
* Centralize Kafka credential handling (in one module) opening the road to more code reuse and better encapsulation;
* Remove kafka dependencies (e.g., kafka-client) from hive-exec module.
* Facilitate unit testing and product stability.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
```
mvn -pl ql test -Dtest=TestDagUtils*
mvn -pl kafka-handler test
```
